### PR TITLE
[UIK-2455][filter trigger] a11y

### PR DESCRIPTION
**Pull request by ilyabrower**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly or it's not required.
- [ ] Unit tests are not broken.
- [ ] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Review by Valeria-Zimnitskaya — APPROVED** [review]

File: semcore/base-trigger/src/FilterTrigger.jsx [hunk 1]
```diff
@@ -188,32 +196,48 @@ class ClearButton extends Component {
...
+        <Hint.Trigger>
+          <SFilterTrigger
+            render={BaseTrigger}
+            tag='button'
+            type='button'
+            size={size}
+            empty={empty}
+            selected
+            disabled={disabled}
+            title={getI18nText('clear')}
```
Comment (sheila-semrush) [inline semcore/base-trigger/src/FilterTrigger.jsx:214]:
```suggestion
            title={getI18nText('clear')}
```
can we replace title with aria-label?

File: semcore/base-trigger/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [4.49.0] - 2024-11-20
+
+### Fixed
+
+- Added `Hint` to the `FilterTrigger.ClearButton`.
+- Remove redundant `aria-label` & `aria-labelledby` from `FilterTrigger.ClearButton`.
+- Added `role` combobox to the `FilterTrigger.TriggerButton`.
+- Added screenreader only `selected` text to the `FilterTrigger.Counter` value.
```
Comment (sheila-semrush) [inline semcore/base-trigger/CHANGELOG.md:12]:
```suggestion
- Added `count` prop to `FilterTrigger.Counter`, along with screenreader only `selected` text
```
to mention the new prop as well

**Review by sheila-semrush — COMMENTED** [review]
Left a comment, otherwise everything's good.

Also:
* I have one question about the Advanced filters example, will ask tomorrow — replaced example with a link
* asked Glasha if it's ok to hide the placeholder — Glasha said ok
* created a task about Dot aria label warning — moved to the next major

File: semcore/base-trigger/CHANGELOG.md [hunk 3]
```diff
@@ -2,1079 +2,1088 @@
...
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [4.49.0] - 2024-11-20
+
+### Fixed
+
+* Added `Hint` to the `FilterTrigger.ClearButton`.
+* Remove redundant `aria-label` & `aria-labelledby` from `FilterTrigger.ClearButton`.
+* Added `role` combobox to the `FilterTrigger.TriggerButton`.
+* Added `count` prop to `FilterTrigger.Counter`, along with screenreader only `selected` text
```
Comment (sheila-semrush) [inline semcore/base-trigger/CHANGELOG.md:12]: let's also mention that placeholder text is now aria-hidden

Comment (ilyabrower) [reply semcore/base-trigger/CHANGELOG.md:12]: added

File: semcore/base-trigger/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,16 @@
...
 
+## [4.49.0] - 2024-11-20
+
+### Fixed
+
+- Added `Hint` to the `FilterTrigger.ClearButton`.
+- Remove redundant `aria-label` & `aria-labelledby` from `FilterTrigger.ClearButton`.
+- Added `role` combobox to the `FilterTrigger.TriggerButton`.
+- Added `count` prop to `FilterTrigger.Counter`, along with screenreader only `selected` text.
+- `placeholder` text is now `aria-hidden` for the `FilterTrigger.TriggerButton` because of `aria-label` and double pronouncing.
```
Comment (sheila-semrush) [inline semcore/base-trigger/CHANGELOG.md:13]: sorry for clinging to this, but maybe better not mention specifically FilterTrigger.TriggerButton? People might think it's only relevant for that component, when it's also relevant for Select.Trigger and other BaseTrigger-based components, if I understand correctly

Comment (ilyabrower) [reply semcore/base-trigger/CHANGELOG.md]: fixed

**Pull request merged**